lib/api/metadataSearch.js

```diff
@@ -0,0 +1,410 @@
+    }
+
+    const escapeXmlFn = listParams.encoding === 'url' ?
+        querystring.escape : escapeForXml;
+    xmlParams.forEach(p => {
+        if (p.value && skipUrlEncoding.has(p.tag)) {
+            xml.push(`<${p.tag}>${p.value}</${p.tag}>`);
+        } else if (p.value || p.tag === 'KeyCount') {
+            xml.push(`<${p.tag}>${escapeXmlFn(p.value)}</${p.tag}>`);
+        } else if (p.tag !== 'NextMarker' &&
+            p.tag !== 'EncodingType' &&
+            p.tag !== 'Delimiter' &&
+            p.tag !== 'StartAfter') {
+            xml.push(`<${p.tag}/>`);
+        }
+    });
+
+    list.Contents.forEach(item => {
+        const v = item.value;
+        if (v.isDeleteMarker) {
+            return null;
+        }
+        const objectKey = escapeXmlFn(item.key);
+        xml.push(
+            '<Contents>',
+            `<Key>${objectKey}</Key>`,
+            `<LastModified>${v.LastModified}</LastModified>`,
+            `<ETag>&quot;${v.ETag}&quot;</ETag>`,
+            `<Size>${v.Size}</Size>`
+        );
+        if (!listParams.v2 || listParams.fetchOwner) {
+            xml.push(
+                '<Owner>',
+                `<ID>${v.Owner.ID}</ID>`,
+                `<DisplayName>${v.Owner.DisplayName}</DisplayName>`,
+                '</Owner>'
+            );
+        }
+        return xml.push(
+            `<StorageClass>${v.StorageClass}</StorageClass>`,
+            '</Contents>'
+        );
+    });
+    list.CommonPrefixes.forEach(item => {
+        const val = escapeXmlFn(item);
+        xml.push(`<CommonPrefixes><Prefix>${val}</Prefix></CommonPrefixes>`);
+    });
+    xml.push('</ListBucketResult>');
+    return xml.join('');
+}
+
+function handleResult(listParams, requestMaxKeys, encoding, authInfo,
+                      bucketName, list, corsHeaders, log, callback) {
+    // eslint-disable-next-line no-param-reassign
+    listParams.maxKeys = requestMaxKeys;
+    // eslint-disable-next-line no-param-reassign
+    listParams.encoding = encoding;
+    let res;
+    if (listParams.listingType === 'DelimiterVersions') {
+        res = processVersions(bucketName, listParams, list,
+            config.versionIdEncodingType);
+    } else {
+        res = processMasterVersions(bucketName, listParams, list);
+    }
+    pushMetric('listBucket', log, { authInfo, bucket: bucketName });
+    monitoring.promMetrics('GET', bucketName, '200', 'listBucket');
+    return callback(null, res, corsHeaders);
+}
+
+/**
+ * bucketGet - Return list of objects in bucket, supports v1 & v2
+ * @param  {AuthInfo} authInfo - Instance of AuthInfo class with
+ *                               requester's info
+ * @param  {object} request - http request object
+ * @param  {function} log - Werelogs request logger
+ * @param  {function} callback - callback to respond to http request
+ *  with either error code or xml response body
+ * @return {undefined}
+ */
+function metadataSearch(authInfo, request, log, callback) {
```

Comment:
Quick note on this, if reviewers are wondering where is the logic calling this function instead of `bucketGet`, it is in Arsenal: https://github.com/scality/Arsenal/commit/68c5b42e6f8994c2d9ba10d91b70d9a01bbcd759#diff-b2fc5116de013086739badfff8f252c51deb78bffd5b24fe117f1edb1fc16b2eR103